### PR TITLE
[mono] Add GC unsafe transition to mono_unhandled_exception 

### PR DESCRIPTION
**Pull request by thaystg**
To fix it: https://github.com/dotnet/runtime/issues/44526

### CODE REVIEW

File: src/mono/mono/metadata/object.c [hunk 1]
```diff
@@ -4148,7 +4148,9 @@ mono_unhandled_exception_internal (MonoObject *exc_raw)
 void
 mono_unhandled_exception (MonoObject *exc)
 {
+	MONO_ENTER_GC_UNSAFE;
 	MONO_EXTERNAL_ONLY_VOID (mono_unhandled_exception_internal (exc));
+	MONO_EXIT_GC_UNSAFE;
```
Comment (lambdageek) [inline src/mono/mono/metadata/object.c:4153]:
```suggestion
	MONO_EXTERNAL_ONLY_GC_UNSAFE_VOID (mono_unhandled_exception_internal (exc));
```

**Review by lambdageek — APPROVED** [review]

File: src/mono/mono/metadata/object.c (hunk 1 above)
Comment (nelsondani) [reply src/mono/mono/metadata/object.c]: I

**Pull request merged**